service/src/main/java/com/expedia/alertmanager/service/dao/AlertStoreService.java

```diff
@@ -102,7 +102,7 @@ public void saveAlerts(List<Alert> alerts) {
         for (final AlertStore store : stores) {
             store.read(labels, from, to, size, (receivedAlerts, ex) -> {
                 synchronized (alerts) {
-                    if (ex == null) {
+                    if (ex.getMessage() == "None") {
```

Comment:
Use `"None".equals(ex.getMessage())` please.

https://www.baeldung.com/java-compare-strings

Reply:
Also, comparing on an exception message is extremely brittle. I would expect to be able to change an exception message in the code anytime without breaking conditional logic. This comment applies throughout.